linker error under clang with atomic funcitons

## Expected Behavior
When building under Clang, the build should succeed.

## Current Behavior
Packages downstream fail due to a linker error with `libfastrtps.so.2.0.0`.
```
--- stderr: rmw_fastrtps_shared_cpp                                                                                         
ld.lld: error: /opt/ros/master/install/lib/libfastrtps.so.2.0.0: undefined reference to __atomic_compare_exchange
ld.lld: error: /opt/ros/master/install/lib/libfastrtps.so.2.0.0: undefined reference to __atomic_load
ld.lld: error: /opt/ros/master/install/lib/libfastrtps.so.2.0.0: undefined reference to __atomic_store
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [test/CMakeFiles/test_security_logging.dir/build.make:112: test/test_security_logging] Error 1
make[1]: *** [CMakeFiles/Makefile2:203: test/CMakeFiles/test_security_logging.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
---
Failed   <<< rmw_fastrtps_shared_cpp	[ Exited with code 2 ]
```

```
--- stderr: test_msgs                                                                                                          
ld.lld: error: /opt/ros/master/install/lib/libfastrtps.so.2.0.0: undefined reference to __atomic_compare_exchange
ld.lld: error: /opt/ros/master/install/lib/libfastrtps.so.2.0.0: undefined reference to __atomic_load
ld.lld: error: /opt/ros/master/install/lib/libfastrtps.so.2.0.0: undefined reference to __atomic_store
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [CMakeFiles/test_action_typesupport_cpp_builds.dir/build.make:127: test_action_typesupport_cpp_builds] Error 1
make[1]: *** [CMakeFiles/Makefile2:149: CMakeFiles/test_action_typesupport_cpp_builds.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
---
```


## System information
Ubuntu 20, Clang 11, `LDFLAGS= -fuse-ld=lld`

- Fast-RTPS version: master
- OS: Ubuntu 210
- ROS2: Master

--- a/cmake/modules/FindThirdpartyBoost.cmake
+++ b/cmake/modules/FindThirdpartyBoost.cmake
@@ -19,6 +19,10 @@ else() # Posix
     set(THIRDPARTY_BOOST_LINK_LIBS ${CMAKE_THREAD_LIBS_INIT} rt)
 endif()
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    list(APPEND THIRDPARTY_BOOST_LINK_LIBS atomic)
+endif()
+
 try_compile(IS_THIRDPARTY_BOOST_OK
         ${CMAKE_BINARY_DIR}
         ${PROJECT_SOURCE_DIR}/thirdparty/boost/test/ThirdpartyBoostCompile_test.cpp

